src/app/applications/multicast/multicast-detail/multicast-detail.component.ts

```diff
@@ -0,0 +1,203 @@
+      .showSimpleDialog()
+      .subscribe((response) => {
+        if (response) {
+          this.multicastService
+            .delete(this.multicast.id)
+            .subscribe((response) => {
+              if (response.status !== 0) {
+                this.showDeletedSnack();
+                this.location.back();
+              } else {
+                this.showFailSnack();
+              }
+            });
+        } else {
+          console.log(response);
+        }
+      });
+  }
+
+  showDeletedSnack(): void {
+    this.snackService.showDeletedSnack();
+  }
+
+  showFailSnack(): void {
+    this.snackService.showFailSnack();
+  }
+  showQueueSnack(): void {
+    this.snackService.showInQueueSnack();
+  }
+  keyPressHexadecimal(event) {
+    // make sure only hexadecimal can be typed in input with adresses.
+    var inp = String.fromCharCode(event.keyCode);
+
+    if (/[a-fA-F0-9]/.test(inp)) {
+      return true;
+    } else {
+      event.preventDefault();
+      return false;
+    }
+  }
+  private handleError(error: HttpErrorResponse) {
+    const errors = this.errorMessageService.handleErrorMessageWithFields(error);
+    this.errorMessages = errors.errorFields;
+    this.errorMessages = errors.errorMessages;
+  }
+
+  clickDownlink() {
+    if (this.validateHex(this.downlink.data)) {
+      this.downlinkService.multicastGet(this.multicast.id).subscribe(
+        (response: any) => {
+          console.log(response)
+          if (response.totalCount > 0) {
+            this.openDownlinkDialog();
+          } else {
+            this.startDownlink();
+          }
+        },
+        (error) => {
+          this.handleError(error);
+          console.log(error);
+        }
+      );
+    }
+  }
+  openDownlinkDialog() {
+    const dialog = this.dialog.open(DownlinkDialogComponent, {});
+
+    dialog.afterClosed().subscribe((result) => {
+      if (result === true) {
+        this.startDownlink();
+        console.log(`Dialog result: ${result}`);
+      }
+    });
+  }
+
+  private startDownlink() {
+    this.errorMessages = [];
+    this.downlinkService
+      .multicastPost(this.downlink, this.multicast.id)
+      .subscribe(
+        (response) => {
+          this.showQueueSnack();
+        },
+        (error) => {
+          this.handleError(error);
+        }
+      );
+  }
+
+  private validateHex(input: string): boolean {
+    const isHexinput = /^[a-fA-F\d]+$/.test(input);
+    let validator = false;
+    if (isHexinput) {
+      validator = true;
+    } else {
+      console.log('test');
+      this.addToErrorMessage('MULTICAST.DOWNLINK.NO-PORT-OR-PAYLOAD');
+      validator = false;
+    }
+    return validator;
+  }
+
+  addToErrorMessage(text: string) {
+    this.translate.get([text]).subscribe((translations) => {
+      this.errorMessages.push(translations[text]);
+    });
+  }
+
+  ngOnDestroy(): void {
+    if (this.deleteDialogSubscription) {
+      this.deleteDialogSubscription.unsubscribe();
```

Comment:
Use [optional chaining](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining) to reduce code.
While not relevant here, you can also chain methods (optional calls). E.g. `.unsubscribe?.()`